Subject: [Fwd: [PATCH] ppc64: Fix nasty typo in PTE freeing code]
From: Benjamin Herrenschmidt <benh@kernel.crashing.org>
To: Olaf Hering <olh@suse.de>

Hi Olaf !

Just to make sure you didn't miss this important one !

Ben.


-----Forwarded Message-----
From: Benjamin Herrenschmidt <benh@ozlabs.au.ibm.com>
To: Andrew Morton <akpm@osdl.org>
Cc: Linus Torvalds <torvalds@osdl.org>, Linux Kernel list <linux-kernel@vger.kernel.org>, Olaf Hering <olh@suse.de>
Subject: [PATCH] ppc64: Fix nasty typo in PTE freeing code
Date: Fri, 07 May 2004 10:40:18 +1000

Hi !

There is a typo in the PTE freeing code causing us to possibly
overflow the batch structure. Please apply this obvious fix:

--- 1.12/include/asm-ppc64/pgalloc.h	Mon Jan 19 17:28:26 2004
+++ edited/include/asm-ppc64/pgalloc.h	Fri May  7 10:38:12 2004
@@ -78,8 +78,8 @@
 	struct page *	pages[0];
 };
 
-#define PTE_FREELIST_SIZE	((PAGE_SIZE - sizeof(struct pte_freelist_batch) / \
-				  sizeof(struct page *)))
+#define PTE_FREELIST_SIZE	((PAGE_SIZE - sizeof(struct pte_freelist_batch)) / \
+				  sizeof(struct page *))
 
 extern void pte_free_now(struct page *ptepage);
 extern void pte_free_submit(struct pte_freelist_batch *batch);
-- 
Benjamin Herrenschmidt <benh@kernel.crashing.org>

